nireports/interfaces/dmri.py

```diff
@@ -0,0 +1,96 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+#
+# Copyright 2023 The NiPreps Developers <nipreps@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+"""Diffusion MRI -specific visualization."""
+import numpy as np
+import nibabel as nb
+
+from nipype.utils.filemanip import fname_presuffix
+from nipype.interfaces.base import (
+    File,
+    BaseInterfaceInputSpec,
+    TraitedSpec,
+    SimpleInterface,
+    traits,
+    isdefined,
+)
+from nireports.reportlets.modality.dwi import plot_heatmap
+
+
+class _DWIHeatmapInputSpec(BaseInterfaceInputSpec):
+    in_file = File(exists=True, mandatory=True, desc="a DWI file")
+    scalarmap = File(exists=True, mandatory=True, desc="a DWI-derived scalar map")
+    mask_file = File(exists=True, desc="a 3D mask")
```

Comment:
Maybe this should be mandatory? Otherwise you'd get a lot of funky stuff coming from non-brain parts of the volume.